azure/scope/cluster.go

```diff
@@ -339,6 +341,9 @@ func (s *ClusterScope) NatGatewaySpecs() []azure.ResourceSpecGetter {
 						Name: subnet.NatGateway.NatGatewayIP.Name,
 					},
 					AdditionalTags: s.AdditionalTags(),
+					// This is a bit of a hack, but we need to know if the VNet is managed to know if
+					// this NAT Gateway was-managed or not.
+					IsVnetManaged: s.IsVnetManaged(),
```

Comment:
For my own understanding: Why is this considered a hack? It seems like the `NatGatewaySpec` has a field for `IsVnetManaged` and follows the convention for the other fields above.

Reply:
I'd argue this is not really a hack.

Reply:
I see that the word `hack` and the way I used it is creating controversy, happy to remove it. 
The context on why I sort of called it a hack is that we created a "new field" to save the answer if this NATGateway was managed. Now that I think of it, it is doing what it is supposed to.

The comment should mostly say 
`// we need to know if the VNet is managed to decide if this NAT Gateway was managed or not.`

Reply:
Updated via 8dc5cc3b4a4fee169b45efee1bc16fe34f91457e